### PR TITLE
kubeadm: remove the cgroup driver detection for docker

### PR DESCRIPTION
**Pull request by neolit123**
#### What this PR does / why we need it:

remove the automatic detection and matching of cgroup drivers for Docker.
Also remove the "IsDockerSystemdCheck" preflight check that checks if the Docker cgroup driver is set to "systemd".
Ideally such detection / coordination should be on the side of CRI implementers and the kubelet (tracked here https://github.com/kubernetes/kubernetes/issues/99808).

Remove the detection during:
- During preflight
- During kubelet config defaulting

#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/kubeadm/issues/874
xref https://github.com/kubernetes/kubernetes/issues/99808

#### Special notes for your reviewer:

NONE

#### Does this PR introduce a user-facing change?

```release-note
ACTION REQUIRED: kubeadm: remove the automatic detection and matching of cgroup drivers for Docker. For new clusters if you have not configured the cgroup driver explicitly you might get a failure in the kubelet on driver mismatch (kubeadm clusters should be using the "systemd" driver). Also remove the "IsDockerSystemdCheck" preflight check (warning) that checks if the Docker cgroup driver is set to "systemd". Ideally such detection / coordination should be on the side of CRI implementers and the kubelet (tracked here https://github.com/kubernetes/kubernetes/issues/99808). Please see the following page on how to configure cgroup drivers with kubeadm manually: https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/configure-cgroup-driver/
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by neolit123** [comment]
dockershim is not adding this detection in 1.21
https://github.com/kubernetes/kubernetes/pull/98357#issuecomment-790949603

perhaps we should just delete it from kubeadm in the next release.
/milestone clear

**Comment by neolit123** [comment]
/cc @pacoxu 
i have updated this PR.

/milestone v1.22

File: cmd/kubeadm/app/constants/constants.go [hunk 1]
```diff
@@ -398,8 +398,6 @@ const (
 
 	// CgroupDriverSystemd holds the systemd driver type
 	CgroupDriverSystemd = "systemd"
-	// CgroupDriverCgroupfs holds the cgroupfs driver type
-	CgroupDriverCgroupfs = "cgroupfs"
```
Comment (neolit123) [inline cmd/kubeadm/app/constants/constants.go:402]: no longer needed.

**Comment by pacoxu** [comment]
Overall LGTM.
As this is something with **action required**, I'd like this PR can be noticed by more reviewers.😄

**Comment by neolit123** [comment]
i realized that we need a work item to also update kinder:
https://github.com/kubernetes/kubeadm/issues/874

>  pin the cgroup driver for docker / containerd to "systemd" in kinder. should be done here; information about setting the driver for CRs is here

otherwise our CI could fail if the kinder node image has cgroupfs in the docker settings.

we should hold this PR until kinder is adapted.

**Comment by pacoxu** [comment]
/cc fabriziopandini
for a look
/lgtm
/hold cancel

**Comment by neolit123** [comment]
@pacoxu 

i put the hold because i think now our CI will start failing without:
https://github.com/kubernetes/kubeadm/issues/874

**Pull request merged**